### ISSUE TITLE
How can I use Visual observation better?

### ISSUE THREAD

**Issue by MarkTension** [Unity-Technologies/ml-agents#3237]
I'm training a network to navigate a 48x48 2D grid, and switch pixels from on to off or off to on. The agent receives a small reward if correct, and small punishment if incorrectly pixel plotted.

At the moment I'm testing to only use a simple visual observation of the entire environmentas input. Unfortunately, it barely trains. When instead using additional vectors as input containing information about nearby pixels' state around the agent, the agent learns the task quite well.

Each step, the agent moves up down left right, and plot pixel or not. The agent is visualized in the environment as a red square with white center dot. (also tried single red pixel, which didn't work either). On-pixels within the red square are colored purple. The network has 512 hidden units.

Is there something I'm obviously doing wrong, or any ways to make the agent learn visual input better?

![screenshot](https://user-images.githubusercontent.com/16321685/72513885-fd357e00-384d-11ea-8d2f-1994e1121fe9.png)
![screenshot](https://user-images.githubusercontent.com/16321685/72513890-00c90500-384e-11ea-8141-972edf122b79.png)

The orange line is the training with only visual observations, the grey one contained vector observations about the immediate neighboring pixel state as well.

Suggestions are very much appreciated:) (let me know if I can supply any more training details)

**Comment by andrewcoh** [comment]
Is that image the agent view?  Also, can you elaborate on what the additional vector observations are that helped?

My first guess is that in the purely visual setting the red agent pixel is blocking the pixel that it's affecting so it cannot observe the true state.  Thus, it would be impossible to know if the pixel the agent was sitting on is in the correct state or not because it is always 'red' in the observation. I may be misunderstanding the problem setting. Please let me know!

**Comment by MarkTension** [comment]
Hey thank you for the response. Everything within the frame is the purple frame is the agent view. 
The additional vectors were a little 3x3 grid around the agent: each pixel being represented as a one-hot encoding of [outside-grid, off-pixel, on-pixel]. 

Your first guess is a problem I had earlier, but I fixed it by either using a single pixel for the agent location (it needs to decide which adjacent pixel it will flip or not flip), or the colors within the agent's red circle would turn purple when pixel was turned on like in the picture.  (I gave agent a bigger circle instead of a single pixel because a single pixel was concerned it was not significant enough for the CNN with the shrinking mapsize after each convolution ).

What over the weekend solved my problem was making the agent's view ego-centric, (keeping the agent centered in the view, mario-style). This was apparently a better representation of the agent's state because I guess it contains more change and diversity in the data, and maybe it's easier to ascribe value to changes of fixed pixels on the screen.
